src/executor/executor.ts

```diff
@@ -454,19 +479,12 @@ export class TaskExecutor {
     });
   }
 
-  public async cancel(reason?: string) {
-    try {
-      if (this.isCanceled) return;
-      if (runtimeContextChecker.isNode) this.removeSignalHandlers();
-      const message = `Executor has interrupted by the user. Reason: ${reason}.`;
-      this.logger.warn(`${message}. Stopping all tasks...`, {
-        tasksInProgress: this.taskQueue.size,
-      });
-      this.isCanceled = true;
-      await this.shutdown();
-    } catch (error) {
-      this.logger.error(`Error while cancelling the executor`, error);
-    }
+  private handleCriticalError(e: Error) {
+    this.options.eventTarget?.dispatchEvent(new Events.ComputationFailed({ reason: e.toString() }));
+    this.logger.error("Critical error", e);
```

Comment:
Consider changing this to a standard console.error to hint the user about the issue even if they didn’t configure logging on TE